src/plugins/unified_search/public/filter_bar/filter_editor/lib/filter_operators.ts

```diff
@@ -127,42 +164,82 @@ const rangeOperatorsSharedProps = {
   },
 };
 
-export const isBetweenOperator = {
+export const isBetweenOperator: Operator = {
   ...rangeOperatorsSharedProps,
   message: strings.getIsBetweenOperatorOptionLabel(),
   id: OPERATORS.BETWEEN,
   negate: false,
+  getParamsFromPrevOperator: (prevOperator, params) => {
+    if (!prevOperator) return;
+    if (
+      [
+        OPERATORS.LESS,
+        OPERATORS.GREATER_OR_EQUAL,
+        OPERATORS.BETWEEN,
+        OPERATORS.NOT_BETWEEN,
+      ].includes(prevOperator.id)
+    )
+      return params;
+  },
+};
+
+export const isNotBetweenOperator: Operator = {
+  ...rangeOperatorsSharedProps,
+  message: strings.getIsNotBetweenOperatorOptionLabel(),
+  negate: true,
+  id: OPERATORS.NOT_BETWEEN,
+  getParamsFromPrevOperator: (prevOperator, params) => {
```

Comment:
This `getParamsFromPrevOperator`  is very smart Marta, I like it

Reply:
Thanks 😊